CRM/SMS/BAO/Provider.php

```diff
@@ -131,6 +137,7 @@ public static function del($providerID) {
 
     $dao = new CRM_SMS_DAO_Provider();
     $dao->id = $providerID;
+    $dao->whereAdd = "(domain_id = " .  CRM_Core_Config::domainID() . "OR domain_id IS NULL)";
```

Comment:
Is there a reason we can't we use the new hook_civicrm_selectWhereClause hook to implement this?